File: Golang/Greedy/BrokenCal.go
package main

import "fmt"

func brokenCalcu(startValue, target int) int {
	res := 0
	for target > startValue {
		if target%2 == 0 {
			target /= 2
		} else {
			target += 1
		}
		res += 1
	}
	return res + (startValue - target)
}

func main() {
	startValue, target := 3, 10
	fmt.Print(brokenCalcu(startValue, target))
}

File: Java/Greedy/BrokenCal.java
public class BrokenCal {
    public static int brokenCalcu(int startValue, int target){
        int res = 0;
        while(target > startValue){
            if(target % 2 == 0){
                target /= 2;
            }else{
                target += 1;
            }
            res += 1;
        }
        return res + (startValue - target);
    }
    public static void main(String[] args){
        
    }
}
